crates/sui-indexer-alt-graphql/src/api/types/available_range.rs

```diff
@@ -46,30 +44,387 @@ impl AvailableRange {
+                    if let Some((delegate_type, delegate_field)) = delegation {
+                        let type_map = type_delegations.entry(type_name.to_string()).or_default();
+
+                        for field in *fields {
+                            let resolved_delegate_field = if *delegate_field == "*" {
+                                field.to_string()
+                            } else {
+                                delegate_field.to_string()
+                            };
+
+                            type_map.insert(
+                                field.to_string(),
+                                (delegate_type.to_string(), resolved_delegate_field)
+                            );
+                        }
+                    }
+                }
+
+                for (interface_name, meta_type) in registry.types.iter() {
+                    let (possible_types, interface_fields) = match meta_type {
+                        MetaType::Interface { possible_types, fields, .. } => (possible_types, fields),
+                        _ => continue,
+                    };
+
+                    for type_name in possible_types {
+                        let type_dels = type_delegations.get(type_name.as_str());
+
+                        for (interface_field_name, _) in interface_fields {
+                            if let Some(delegations) = type_dels {
+                                if let Some((delegate_type, delegate_field)) = delegations.get(interface_field_name) {
+                                    assert_eq!(
+                                        delegate_type, interface_name,
+                                        "Type '{}' field '{}' should delegate to interface '{}' but delegates to '{}'",
+                                        type_name, interface_field_name, interface_name, delegate_type
+                                    );
+                                    assert_eq!(
+                                        delegate_field, interface_field_name,
+                                        "Type '{}' field '{}' should delegate to interface field '{}' but delegates to '{}'",
+                                        type_name, interface_field_name, interface_field_name, delegate_field
+                                    );
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+
+            /// Validates that the macro invocation matches the schema. This is to catch any typos in types or fields in the macro invocation.
+            fn test_macro_invocation_matches_schema(registry: &Registry) {
+                let mut type_field_filters = Vec::new();
+                $(
+                    let type_name = stringify!($type);
+                    let field_names = vec![$(stringify!($field)),*];
+                    type_field_filters.push((type_name, field_names));
+                )*
```

Comment:
This is another situation where you can have the macro generate some test-only data as a `const` and then reference it in a test written outside the macro.